### ISSUE TITLE
hb.data.ParquetDataset in hb.estimator.train_and_evaluate will loss data

### ISSUE THREAD

**Issue by karterotte** [alibaba/HybridBackend#152]
# Current behavior
I have a parquet contain 300 records.
Using hb.data.ParquetDataset will be OK shows 300 records
![image](https://github.com/DeepRec-AI/HybridBackend/assets/9114743/61ebd934-7951-4155-a707-633ad565d440)
But When I'm using hb.estimator.train_and_evaluate to train a model, It shows only 100 records
![image](https://github.com/DeepRec-AI/HybridBackend/assets/9114743/f732ac88-be89-4619-b737-24680ff0e972)

When I set eval_every_n_iter=None,It seems OK(300 records)
Maybe sth wrong in eval dataset reading

# System information
- GPU model and memory:
- OS Platform:
- Docker version: alideeprec/deeprec-release:deeprec2306-gpu-py38-cu116-ubuntu20.04-hybridbackend
- GCC/CUDA/cuDNN version:
- Python/conda version:
- TensorFlow/PyTorch version:  tf1.15,deeprec

# Code to reproduce

```python
# Wrong
hb.estimator.train_and_evaluate(
        estimator, train_spec, eval_spec, eval_every_n_iter=100)
# OK
hb.estimator.train_and_evaluate(
        estimator, train_spec, eval_spec, eval_every_n_iter=None)
```

# Willing to contribute

Yes

**Comment by karterotte** [comment]
my bug：eval used train loss,it will cost train data
